packages/gensx/src/resolve.ts

```diff
@@ -53,5 +53,8 @@ export async function resolveDeep<T>(value: unknown): Promise<T> {
  * This is the main entry point for executing workflow components.
  */
 export async function execute<T>(element: ExecutableValue): Promise<T> {
-  return (await resolveDeep(element)) as T;
+  const context = getCurrentContext().getWorkflowContext();
+  const result = (await resolveDeep(element)) as T;
+  context.checkpointManager.write();
```

Comment:
If you don't await this, we might lose it if the program immediately ends right? I think this should probably be an awaitable `flush`.

Reply:
In theory. Not if there is a `process.exit` kicking around, or if its in a serverless environment.

[Langsmith](https://docs.smith.langchain.com/observability/how_to_guides/tracing/serverless_environments) provides an env var for this, [Sentry](https://docs.sentry.io/platforms/javascript/guides/aws-lambda/configuration/draining/) suggests you need to manually call `.close`.